### DISCUSSION TITLE
How to use FAIRSEQ Models on COQUI server command?

### DISCUSSION THREAD

**Discussion by prashantpatole** [coqui-ai/TTS#3211] [General Q&A]
Platform: Windows 10, Visual Studio Code, .venv, 3060 Ti, New to TTS

 I wish to TTS to native Indian languages (here example, Marathi, with iso mar) using models downloaded from https://github.com/facebookresearch/fairseq/tree/main/examples/mms#asr
MMS-1B:FL102	102	FLEURS

I tried the steps given in [https://stackoverflow.com/questions/66726331/how-can-i-run-mozilla-tts-coqui-tts-training-with-cuda-on-a-windows-system](https://stackoverflow.com)

However, COQUI has also been updated to the latest Python, and there are many version conflicts. 
I have started fresh and got the latest COQUI running on VSCode.

Please help if I am missing anything. Please also help to get CUDA.  

1. I could execute default models in COQUI on the CPU (not on CUDA). However, I could not run the models not listed in .model.json.

2. I have tried adding/removing this to .model.json inside section '"tts_models": {'
```
,
        "mar": {
            "fairseq": {
                "vits":{
                    "description": "marathi fair seq.",
                    "github_rls_url":"https://dl.fbaipublicfiles.com/mms/tts/mar.tar.gz",
                    "default_vocoder": "null",
                    "commit": "da8fb63",
                    "license": "CC-BY-SA 4.0",
                    "contact": "pratap2023mms"
                }
            }
        }
```
 3. This is the command I am using to start the server.
`python .\TTS\server\server.py --model_name tts_models/mar/fairseq/vits  --model_path F:\Docker\TTL\tts_models\tts_models--mar--fairseq--vits\  --debug true --show_details true --config_path  "F:\Docker\TTL\tts_models\tts_models--mar--fairseq--vits\config.json" 
`
4. Followed this https://github.com/coqui-ai/TTS#example-text-to-speech-using-fairseq-models-in-1100-languages-
but this doesn't mention about server command.

I am stuck on this error.
`Traceback (most recent call last):
  File "F:\Docker\fairseq\src\COQUI-TTS\TTS\server\server.py", line 104, in <module>
    synthesizer = Synthesizer(
  File "F:\Docker\fairseq\src\COQUI-TTS\TTS\utils\synthesizer.py", line 93, in __init__
    self._load_tts(tts_checkpoint, tts_config_path, use_cuda)
  File "F:\Docker\fairseq\src\COQUI-TTS\TTS\utils\synthesizer.py", line 183, in _load_tts
    self.tts_config = load_config(tts_config_path)
  File "F:\Docker\fairseq\src\COQUI-TTS\TTS\config\__init__.py", line 99, in load_config
    model_name = _process_model_name(config_dict)
  File "F:\Docker\fairseq\src\COQUI-TTS\TTS\config\__init__.py", line 67, in _process_model_name
    model_name = model_name.replace("_generator", "").replace("_discriminator", "")
AttributeError: 'dict' object has no attribute 'replace'`

**Comment by erogol** [comment]
I don't think the server supports those. You can try running it with the right model name. This can help -> https://tts.readthedocs.io/en/latest/inference.html#example-text-to-speech-using-fairseq-models-in-1100-languages
